kornia/core/module.py

```diff
@@ -0,0 +1,273 @@
+        if isinstance(
+            output_image,
+            (
+                list,
+                tuple,
+            ),
+        ):
+            return type(output_image)([self._detach_tensor_to_cpu(out) for out in output_image])
+        raise RuntimeError
+
+    def show(self, n_row: Optional[int] = None, backend: str = "pil") -> "Image.Image":
+        """Returns PIL images.
+
+        Args:
+            n_row: Number of images displayed in each row of the grid.
+            backend: visulization backend. Only PIL is supported now.
+        """
+        if self._output_image is None:
+            raise ValueError("No pre-computed images found. Needs to execute first.")
+        if backend.lower() not in ["pil"]:
+            raise ValueError(f"Unsupported backend `{backend}`.")
+
+        if len(self._output_image.shape) == 3:
+            out_image = self._output_image
+        elif len(self._output_image.shape) == 4:
+            if n_row is None:
+                n_row = math.ceil(self._output_image.shape[0] ** 0.5)
+            out_image = kornia.utils.image.make_grid(self._output_image, n_row, padding=2)
+        else:
+            raise ValueError
+        return Image.fromarray((out_image.permute(1, 2, 0).squeeze().numpy() * 255).astype(np.uint8))
+
+    def save(self, name: Optional[str] = None, n_row: Optional[int] = None) -> None:
+        """Saves the output image(s) to a directory.
+
+        Args:
+            name: Directory to save the images.
+            n_row: Number of images displayed in each row of the grid.
+        """
+        if name is None:
+            name = f"Kornia-{datetime.datetime.fromtimestamp(time.time()).strftime('%Y%m%d%H%M%S')!s}.jpg"
+        if len(self._output_image.shape) == 3:
+            out_image = self._output_image
+        if len(self._output_image.shape) == 4:
+            if n_row is None:
+                n_row = math.ceil(self._output_image.shape[0] ** 0.5)
+            out_image = kornia.utils.image.make_grid(self._output_image, n_row, padding=2)
+        kornia.io.write_image(name, out_image.mul(255.0).byte())
+
+
+class ImageModule(Module, ImageModuleMixIn):
+    """Handles image-based operations.
+
+    This modules accepts multiple input and output data types, provides end-to-end
+    visualization, file saving features. Note that this module fits the classes that
+    return one image tensor only.
+
+    Note:
+        The additional add-on features increase the use of memories. To restore the
+        original behaviour, you may set `disable_features = True`.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._disable_features: bool = False
+
+    @property
+    def disable_features(self):
+        return self._disable_features
+
+    @disable_features.setter
+    def disable_features(self, value: bool = True):
+        self._disable_features = value
+
+    def __call__(
```

Comment:
we should test all this against onnx and dynamo / compile

Reply:
I tested with JIT, not sure about the others.

Reply:
+1, since it'll go as module for all things

Reply:
is the dynamo tests in the CI enough?

Reply:
yes

Reply:
Then it should be good now.